src/MaxMind/Db/Reader/Decoder.php

```diff
@@ -286,17 +286,13 @@ private function decodePointer(int $ctrlByte, int $offset): array
                 // machines
                 $pointerOffset = $this->decodeUint($buffer, $pointerSize);
 
-                $byteLength = $pointerSize + $this->pointerBaseByteSize;
+                $pointerBase = $this->pointerBase;
 
-                if ($byteLength <= _MM_MAX_INT_BYTES) {
+                if (\PHP_INT_MAX - $pointerBase >= $pointerSize) {
                     $pointer = $pointerOffset + $this->pointerBase;
```

Comment:
Should we use `$pointerBase` here since we have it? (As opposed to `$this->pointerBase`)